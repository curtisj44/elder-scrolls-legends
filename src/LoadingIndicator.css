.loading-indicator {
  align-items: center;
  background: linear-gradient(0, rgba(0, 0, 0, .5) 15%, rgba(0, 0, 0, 0)  100%);
  bottom: 0;
  display: flex;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
}

.loading-indicator__text {
  background: var(--dark-moon);
  border: 2px solid var(--bleach-white);
  box-shadow: 0 5px 20px rgba(0, 0, 0, .5);
  margin: 60px 0 30px;
  padding: 2em 4em;
}
