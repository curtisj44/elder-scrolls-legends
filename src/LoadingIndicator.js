// Dependencies
// ------------

// Libraries
import React from 'react';
// Styles
import './LoadingIndicator.css';

// External
// --------

// TODO: probably need some ARIA

export default () => (
  <div className="loading-indicator">
    <div className="loading-indicator__text">Loading&hellip;</div>
  </div>
);
