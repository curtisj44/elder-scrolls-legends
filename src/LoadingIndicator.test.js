// Dependencies
// ------------

// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import LoadingIndicator from './LoadingIndicator';

// Setup
// -----

const component = renderer.create(<LoadingIndicator />);

// Tests
// -----

describe('LoadingIndicator', () => {
  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
