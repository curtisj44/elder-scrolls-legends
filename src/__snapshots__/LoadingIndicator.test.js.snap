// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator renders correctly 1`] = `
<div
  className="loading-indicator"
>
  <div
    className="loading-indicator__text"
  >
    Loadingâ€¦
  </div>
</div>
`;
